server/src/main/java/org/apache/cloudstack/cluster/ClusterDrsServiceImpl.java

```diff
@@ -357,10 +372,90 @@ List<Ternary<VirtualMachine, Host, Host>> getDrsPlan(Cluster cluster,
                     serviceOfferingDao.findByIdIncludingRemoved(vm.getId(), vm.getServiceOfferingId()));
         }
 
+        // PHASE 1: Pre-compute suitable hosts (once per eligible VM - never changes)
+        // Use listHostsForMigrationOfVM to get hosts validated by getCapableSuitableHosts
+        // This ensures DRS uses the same validation as "find host for migration" command
```

Comment:
new method with this as javadoc;
```suggestion
    /*
     * PHASE 1: Pre-compute suitable hosts (once per eligible VM - never changes)
     *   Use listHostsForMigrationOfVM to get hosts validated by getCapableSuitableHosts
     *   This ensures DRS uses the same validation as "find host for migration” command
     */
```